src/main/java/fi/metatavu/metamind/server/rest/StoriesApiImpl.java

```diff
@@ -160,20 +190,48 @@ public Response createIntent(Intent body, UUID storyId) {
     IntentTrainingMaterial intentRegexTrainingMaterial = trainingMaterialController.setIntentTrainingMaterial(intent, TrainingMaterialType.INTENTREGEX, intentRegexMaterial);
     IntentTrainingMaterial variableOpenNlpRegexTrainingMaterial = trainingMaterialController.setIntentTrainingMaterial(intent, TrainingMaterialType.VARIABLEOPENNLPREGEX, variableOpenNlpRegexMaterial);
     
+    try {
+      UUID createdResourceId = authenticationController.createProtectedResource(loggedUserId, String.format(INTENT_NAME_BEGINNING, intent.getId()), String.format("/v2/stories/intents/%s", intent.getId()), "intent", SCOPES);
```

Comment:
Permissions should be defined only on story level, intent permissions are way too fine grained